exonum/src/runtime/rust/service.rs

```diff
@@ -137,6 +138,43 @@ where
     }
 }
 
+/// Simplifies creation of supplementary entities for `ServiceFactory`.
+pub trait InstanceInfoProvider: ServiceFactory {
```

Comment:
Also in my opinion this name is a little bit unclear. And it seems that you can use `ServiceFactory` trait directly, this would free the user from having to write an empty implementation of this trait.

Reply:
Yeap, I've already implemented `InstanceInfoProvider` for `ServiceFactory`, so no need for manual implementations.
About naming - I'm always open for suggestions 🙂